ivy/functional/frontends/paddle/nn/functional/loss.py

```diff
@@ -43,3 +45,49 @@ def binary_cross_entropy_with_logits(
         ret = ivy.multiply(weight, ret)
     ret = reduction(ret).astype(label.dtype)
     return paddle.to_tensor(ret.reshape([-1]))
+
+
+@handle_exceptions
+@to_ivy_arrays_and_back
+@with_supported_dtypes({"2.4.2 and below": ("float32", "float64")}, "paddle")
+def cosine_embedding_loss(
+    input1, input2, label, margin=0.0, reduction="mean", name=None
+):
+    if len(label.shape) != 1:
+        raise ValueError("1D target tensor expected, multi-target not supported")
+
+    if input1.shape != input2.shape:
+        raise ValueError(
+            "the shape of input tensor 1 should be equal to input tensor 2, but found"
+            " inputs with different sizes"
+        )
+
+    if len(input1.shape) > 2:
+        raise ValueError(
+            "1D target tensor expects 1D or 2D input tensors, but found inputs with"
+            " different sizes"
+        )
+
+    prod_sum = (input1 * input2).sum(axis=-1)
+    mag_square1 = ivy.square(input1).sum(axis=-1) + 10e-12
```

Comment:
For proper notation we can use
```suggestion
    mag_square1 = ivy.square(input1).sum(axis=-1) + 1e-12
```

Reply:
Won't this intefere with the final values ?

Reply:
It won’t interfere as this is epsilon value and the notation should be scientific😄